lib/kafka-consumer.js

```diff
@@ -102,25 +102,35 @@ Consumer.prototype._onWrapper = function (consumerInstance, eventName, cb) {
   }
 
   return consumerInstance.__kafkaAvro_on('data', function(message) {
+    if (!this.sr.keySchemas[message.topic]) {
+      log.warn('_onWrapper() :: Warning, consumer did not find topic on SR for key schema:',
+        message.topic);
+
+      message.parsedKey = message.key
+        ? JSON.parse(message.key.toString('utf-8'))
```

Comment:
this will breaks every client that use key as pure string - instead of a unserialized avro key.

But even removing `JSON.parse` and using `message.key.toString('utf-8')`, the `parsedKey` always is null, so this line maybe can be removed (if the producer/consumer wants to use the key, use the size `key` directly.

This also evidences the actual `message.parsed = JSON.parse(message.value.toString('utf-8'));` is very fragile too.

Reply:
I had the same thoughts, because usually the `key` is sent as a `string` or `number`, but Kafka doesn't care so it could be an `Object`, `Buffer`, whatever.  
In the cases where we can't find a schema, I thought it could be so that if the `key` is an `Object`, then it `JSON.parse`s it. Otherwise, just uses the raw `key`.  
The same would apply to `value`, but then it would maybe make things less consistent.  
I decided to just `JSON.parse` the `key` because that's what happens to the `value`.  
But if you agree we can change it to only do that if it is an `Object`.  
And this behaviour would need to be the same for `key` and `value` to keep it as consistent as possible.

Reply:
Hmmm... We have two options here:
- When the key or value doesn't exists in schema, parsedValue/parseKey should be null (and `parsed` is removed). This will breaks API, so this is a major release.
- Keep the functionality of `parsed` actual as is (with this name), create a `parsedKey` with null value when not found, and we can release it as minor release, because it won't breaks anything.

What do you think?

Reply:
I personally don't like breaking APIs 😛 , so I would lean towards the second option.  
If you are okay with it, I can adjust the PR.    
 
And what do you think about the `key` and `value` in cases where the schema is not found being sent as raw when the type is not `Object`, and `JSON.parse` it when it is an `Object`? This way won't break people using pure string for the `key`.

Reply:
Sounds great :)

Reply:
@ricardohbin Could you test now? I made the changes.  
  
Also, is `_transformAvro` in the `Consumer` ever used? Being underscored I believe it was supposed to be a private function, but I can't find any usage anywhere. And it is almost a duplicate of the event 'data' wrapper. Should I remove it?

Reply:
I will check the changes now (sorry for my delay). If everything go right, we can dump the version. About `_transformAvro` you can remove it, or put the call in the callback of event handler 
`return consumerInstance.__kafkaAvro_on('data', this._transformAvro)`

Reply:
No problem. I will remove the duplicate code on `_transformAvro` tomorrow.

Reply:
I have removed it :smile: